canonical-kubernetes/addons/jfrog/steps/01_install_arti/jfrog_common.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+set -eux
+
+. "$CONJURE_UP_SPELLSDIR/sdk/common.sh"
+
+function re_install_helm() {
```

Comment:
@johnsca can any of the latest helm work be applied here?

Reply:
It's definitely worth looking at [the updated `install_helm` helper](https://github.com/conjure-up/spells/blob/master/canonical-kubernetes/addons/common.sh#L7), as it will always reinstall Helm once per deployment, which I think addresses the concern for the force reinstall here, and also the key indicating that the install was already done for the deployment was changed to support that.  There's also some support for Darwin as a platform there.

But init wait here is a lot more forgiving, so we would probably want to fold that from here into the shared helper regardless.

Reply:
Helm re-install as part of the Artifactory add-on is an optional step. 

Is there a way to be 100% sure that tiller ($version) is up before the add-on is executed? Also, can an end user pick the $version of helm to be installed? We did touch this topic before of having a common place to install tools and add-on just relies on this common stuff. And at the same time, make sure that multiple add-ons can co-exist. It's good to know that this implementation is in progress.

But to support this model, and given the nature of changes, we need to play with the "Capabilities" object of Helm to make sure that Artifactory is installed (via helm) on a supported k8s that has a supported version of tiller.